### PR TITLE
Add type definitions to interpolate-components

### PR DESCRIPTION
**Pull request by santios**
#### Changes proposed in this Pull Request

* This patch adds the type definitions for the `interpolate-components` package.
* Updates README to reflect implementation.

#### Testing instructions

No changes on behavior.

### CODE REVIEW

File: packages/interpolate-components/package.json [hunk 1]
```diff
@@ -22,6 +22,7 @@
 	"homepage": "https://github.com/Automattic/wp-calypso/tree/HEAD/packages/interpolate-components#readme",
 	"devDependencies": {
 		"@automattic/calypso-typescript-config": "workspace:^",
+		"@types/react": "^17.0.37",
```
Comment (scinos) [inline packages/interpolate-components/package.json]: I think `@types/*` dependencies are better as `dependencies` (right @sirreal ?)

**Review by scinos — APPROVED** [review]

File: packages/interpolate-components/package.json (hunk 1 above)
Comment (sirreal) [reply packages/interpolate-components/package.json]: `@types/*` dependencies should mirror the dependency they provide types for. If `react` is a `dependency`, its types should also be a `dependency`. In this case, `react` is a _peer dependency_, so **`@types/react` should be a `_peer dependency_**.

**We should also match major versions.** In this case we're depending on `react@>=16.2.0`, the types should be similar. [It looks like the 16.1 types include the definition for `Fragment`](https://unpkg.com/browse/@types/react@16.1.0/index.d.ts), which was the requirement for interpolate components. I'd recommend using `@types/react@>=16.1.0` or `16.3.0` (there's no `16.2`). As far as I can tell `Fragment` was added somewhere between 16.0 and 16.1 in the types.

---

The reasoning for all this is that the types we'll be publishing include `JSX.Element`, so any consumers will see that. If they don't have a definition for `JSX.Element` is, they'll experience errors.

We might also [mark the dependency as optional](https://docs.npmjs.com/cli/v8/configuring-npm/package-json#peerdependenciesmeta). Folks using TypeScript need the dependency for the type system to work, but folks not using TypeScript have absolutely no need for `@types/react`.

Something like the following seems ideal to me:

```json
{
	"peerDependencies": {
		"react": ">=16.2.0",
		"@types/react": ">=16.1.0"
	},
	"peerDependenciesMeta": {
		"@types/react": {
			"optional": true
		}
	},
}
```

Comment (santios) [reply packages/interpolate-components/package.json]: Thanks @sirreal. I applied your suggestion. What's the publishing process @scinos? Thank you!

Comment (scinos) [reply packages/interpolate-components/package.json]: The process I follow is https://github.com/Automattic/wp-calypso/blob/trunk/docs/monorepo.md#publishing

**Comment by sirreal** [comment]
I'll land this and publish the next version.

File: packages/interpolate-components/CHANGELOG.md [hunk 2]
```diff
@@ -0,0 +1,16 @@
+# Releases
```
Comment (scinos) [inline packages/interpolate-components/CHANGELOG.md]: super nit: We could follow the template https://keepachangelog.com/en/1.0.0/. I personally like it adds a heading level with `Added`, `Removed`...

That being said, most of our Changelogs don't follow it anyway, so feel free to ignore :)

Comment (scinos) [reply packages/interpolate-components/CHANGELOG.md]: I'll try to find some time and port existing changelogs to that format, so at least it is unified across the whole repo.

Comment (sirreal) [reply packages/interpolate-components/CHANGELOG.md]: You're right and I agree with you. I'll go the extra _extra_ mile and using that now that my tests have finally passed and I could easily ship this 😆

**Pull request merged**